### DISCUSSION TITLE
RabbitMq #5.20.0 - getSocketFactory() returns null

### DISCUSSION THREAD

**Discussion by mageshmani18** [rabbitmq/rabbitmq-server#10290] [Questions] [answered]
Hello,

I updated rabbitmq (com.rabbitmq:amqp-client) version from 2.8.7 to 5.20.0, and I noticed that the getSocketFactory() method now returns null, resulting in a runtime exception.

Would you please help with what I need to do to fix it? 

Thanks,
Magesh M

**Comment by acogoluegnes** [comment, marked as answer]
What are you doing with the `SocketFactory`? It's likely you just need to use `SocketFactory.getDefault()`.

**Reply by mageshmani18** [reply]
Thank you!

I used this for setting up the socket factory before establishing the connection.
